hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/TrashOzoneFileSystem.java

```diff
@@ -362,10 +358,35 @@ private abstract class OzoneListingIterator implements Closeable {
       if (status.isDirectory()) {
         this.pathKey = addTrailingSlashIfNeeded(pathKey);
       }
-      keyIterator = ozoneManager.getMetadataManager().getKeyIterator();
+      OFSPath fsPath = new OFSPath(pathKey,
+          OzoneConfiguration.of(getConf()));
+      keyIterator =
+          getKeyIterator(fsPath.getVolumeName(), fsPath.getBucketName(),
+              fsPath.getKeyName());
     }
 
-      /**
+    private Iterator<OmKeyInfo> getKeyIterator(String volumeName,
+        String bucketName, String keyName) throws IOException {
+      OMMetadataManager metadataManager = ozoneManager.getMetadataManager();
+      List<OmKeyInfo> keys = new ArrayList<>(
+          metadataManager.listKeys(volumeName, bucketName, "", keyName,
+              OZONE_MAX_LIST_KEYS_SIZE));
+      String lastKey = keys.get(keys.size() - 1).getKeyName();
+      List<OmKeyInfo> nextBatchKeys =
+          metadataManager.listKeys(volumeName, bucketName, lastKey, keyName,
+              OZONE_MAX_LIST_KEYS_SIZE);
+
+      while (!nextBatchKeys.isEmpty()) {
+        keys.addAll(nextBatchKeys);
```

Comment:
Whether it will consume high memory if number of keys are large, now any impact with that?